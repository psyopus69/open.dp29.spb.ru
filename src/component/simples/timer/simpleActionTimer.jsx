import React, {Fragment} from "react";
import {PropTypes} from "prop-types";

// Simple for HOC withTimer
const SimpleActionTimer = (props) => {
  const {clearTimer, setTimer} = props;
  let tmpForInc = 0;
  return <Fragment>
    <div>Test 1.3 SimpleActionTimer React</div>
    <div>Action incTimerStart <button onClick={() => {
      setTimer(() => {
        // eslint-disable-next-line no-console
        console.log(`from timer tmpForInc = ${tmpForInc}`);
        tmpForInc += 1;
      }, 1000);
    }}>startConsoleLog</button><button onClick={() => {
      clearTimer();
    }}>stopIncrement</button> ${`->`} : {tmpForInc}</div>
  </Fragment>;
};
SimpleActionTimer.propTypes = {
  setTimer: PropTypes.func.isRequired,
  clearTimer: PropTypes.func.isRequired,
};

export {
  SimpleActionTimer,
};
