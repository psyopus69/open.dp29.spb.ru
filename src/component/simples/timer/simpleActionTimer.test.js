import React from "react";
import renderer from "react-test-renderer";
import {SimpleActionTimer} from "./simpleActionTimer";

it(`panel with setTimerAction & clear it`, () => {
  const tree = renderer.create(<SimpleActionTimer
    setTimer={jest.fn()}
    clearTimer={jest.fn()}
  />);
  expect(tree).toMatchSnapshot();
});
