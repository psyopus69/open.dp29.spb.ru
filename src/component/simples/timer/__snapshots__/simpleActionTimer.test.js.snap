// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel with setTimerAction & clear it 1`] = `
Array [
  <div>
    Test 1.3 SimpleActionTimer React
  </div>,
  <div>
    Action incTimerStart 
    <button
      onClick={[Function]}
    >
      startConsoleLog
    </button>
    <button
      onClick={[Function]}
    >
      stopIncrement
    </button>
     $
    -&gt;
     : 
    0
  </div>,
]
`;
